core/src/target_os_check.rs

```diff
@@ -0,0 +1,329 @@
+
+    static INIT: Once = Once::new();
+
+    fn init_log() {
+        INIT.call_once(|| {
+            flexi_logger::Logger::try_with_env()
+                .unwrap()
+                .format(
+                    |write: &mut dyn Write, _now: &mut DeferredNow, record: &Record<'_>| {
+                        let file_name = record.file().unwrap_or_default();
+                        let file_name = if file_name.len() > 15 {
+                            let split = file_name.len() - 15;
+                            &file_name[split..]
+                        } else {
+                            file_name
+                        };
+                        write!(
+                            write,
+                            "{file_name:>15}{:>5} - {}",
+                            record.line().unwrap_or_default(),
+                            record.args()
+                        )
+                    },
+                )
+                .start()
+                .unwrap();
+        })
+    }
+
+    #[test]
+    fn test_target_os_nested_reject() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(all(feature = "my-feature", not(target_os = "ios")))]
+            pub struct NestedNotTarget;
+        };
+
+        assert!(!accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_target_os_accept() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(target_os = "android")]
+            pub struct NestedNotTarget;
+        };
+
+        assert!(accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_target_os_combined_any_accepted() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(any(target_os = "android", target_os = "ios"))]
+            pub struct NestedNotTarget;
+        };
+
+        assert!(accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_target_os_combined_all_accepted() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(all(target_os = "windows", target_os = "android"))]
+            pub struct NestedNotTarget;
+        };
+
+        assert!(accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
+
+    #[test]
+    fn test_target_os_combined_rejected() {
+        init_log();
+
+        let test_struct: ItemStruct = parse_quote! {
+            #[cfg(not(any(target_os = "wasm32", target_os = "ios")))]
+            pub struct NestedNotTarget;
+        };
+
+        assert!(!accept_target_os(
+            &test_struct.attrs,
+            &["ios".into(), "android".into()]
+        ));
+    }
```

Comment:
Similarly with this test, if we're running TypeShare with multiple `--target_os` settings, my intuitive thought is that we'll be using the types in _each_ of the output OSes. In that case, running the build for `android` would pass this check, because it is neither `wasm32` nor `ios`. However, this test indicates that it's a rejection, because of the overlap of having `--target_os ios,android`.

I don't have enough of a grasp of the underlying use-case for multiple `target_os` to make an argument for one approach or the other, however given that it's somewhat confusing, it would be beneficial to clearly document the logic that we use to determine whether a type is included or not.